iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -93,10 +93,27 @@ expected<SharedMemoryObject, SharedMemoryObjectError> SharedMemoryObjectBuilder:
         return error<SharedMemoryObjectError>(SharedMemoryObjectError::SHARED_MEMORY_CREATION_FAILED);
     }
 
+    const auto real_size = sharedMemory->get_size();
+    if (!real_size)
+    {
+        printErrorDetails();
+        IOX_LOG(ERROR) << "Unable to create SharedMemoryObject since we could not acquire the memory size of the "
+                          "underlying object.";
+        return error<SharedMemoryObjectError>(SharedMemoryObjectError::UNABLE_TO_VERIFY_MEMORY_SIZE);
+    }
+
+    if (*real_size < m_memorySizeInBytes)
```

Comment:
Nitpicking
```suggestion
    const auto realSizeResult = sharedMemory->get_size();
    if (!realSizeResult)
    {
        printErrorDetails();
        IOX_LOG(ERROR) << "Unable to create SharedMemoryObject since we could not acquire the memory size of the "
                          "underlying object.";
        return error<SharedMemoryObjectError>(SharedMemoryObjectError::UNABLE_TO_VERIFY_MEMORY_SIZE);
    }
    
    const auto realSize = *realSizeResult;
    if (realSize < m_memorySizeInBytes)
```